8.frontend-react/crm-frontend/src/components/Form.jsx

```diff
@@ -0,0 +1,97 @@
+// import React from 'react';
+import '../styles/formStyle.css';
+import React, { useState, useEffect } from 'react';
+import FormField from './FormField';
+import CrmButton from './CrmButton';
+import validate from '../helpers/validationHelper';
+
+
+function Form(props) {
+
+    const [fields, setFields] = useState(props.fields);
+    const [mainError, setMainError] = useState('');
+    const [isLoading, setIsLoading] = useState(false);
+    
+    useEffect(() => {
+        setFields(props.fields);
+    }, [props.fields])
+
+    const setValue = (key, value) => {
+        const fieldsTemp = {...fields};
+        fieldsTemp[key].value = value;
+        fieldsTemp[key].error = false;
+        setFields(fieldsTemp);
+    };
+
+    const submit = async () => {
+        // set true
+        let data = {};
+        const fieldsTmp = {...fields};
+        let validationRes = true;
+
+        for(let key in fields){
+            data[`${key}`] = {}
```

Comment:
no need to wrap key with `${}`
you can use `data[key] = x`
`fieldsTmp[key].error = true;`
etc